python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -262,6 +263,28 @@ def _get_ray_resources_from_group_spec(
     if num_gpus is not None:
         resources["GPU"] = num_gpus
 
+    if num_tpus is not None:
+        """Add TPU head resource, similar to the GCP node_provider.
```

Comment:
I remember that we set `resources["TPU"]` before, didn't we?

Reply:
It was previously set, but removed here after our discussion: https://github.com/ray-project/ray/commit/be3af8e8c8a9e9cce60db9c23ebb391800ad8e00. The git history is messed up due to a bad rebase.

Reply:
Added back in with [3b51219](https://github.com/ray-project/ray/pull/45105/commits/3b51219c26d5af0bc2a594954747aaefa4d8d578).